src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    const copy = [...this];
```

Comment:
for what reason did you make the copy of current array?
check do you really need it